### PR TITLE
[K8s-Array] Add finalizer

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR

- Add finalizer
- Update log-name to be in sync with  actual pod names i.e. pod-${retry}-${index}

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by anandswaminathan** [comment]
+1

**Comment by anandswaminathan** [comment]
PTAL @kumare3 @EngHabu

File: go/tasks/plugins/array/k8s/monitor.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func FetchPodStatusAndLogs(ctx context.Context, client core.KubeClient, name k8s
 			o, err := logPlugin.GetTaskLogs(tasklog.Input{
 				PodName:          pod.Name,
 				Namespace:        pod.Namespace,
-				LogName:          fmt.Sprintf(" #%d-%d", index, retryAttempt),
+				LogName:          fmt.Sprintf(" #%d-%d", retryAttempt, index),
```
Comment (EngHabu) [inline go/tasks/plugins/array/k8s/monitor.go:227]: Don't they get ordered by the name? is this better UX than essentially grouping the task by the index?

File: go/tasks/plugins/array/k8s/task.go [hunk 2]
```diff
@@ -227,8 +251,34 @@ func (t Task) Finalize(ctx context.Context, tCtx core.TaskExecutionContext, kube
...
+	pod := &v1.Pod{
+		TypeMeta: metaV1.TypeMeta{
+			Kind:       PodKind,
+			APIVersion: v1.SchemeGroupVersion.String(),
+		},
+	}
+
+	err := kubeClient.GetClient().Get(ctx, k8sTypes.NamespacedName{
+		Name:      podName,
+		Namespace: tCtx.TaskExecutionMetadata().GetNamespace(),
```
Comment (EngHabu) [inline go/tasks/plugins/array/k8s/task.go:263]: [line 233 above](https://github.com/flyteorg/flyteplugins/pull/186/files#diff-01ee392f44bbd91e1e5f4afd999327e02f8f1df54fd5b5b576478e903cbcb3c2R233) uses a different piece of code to get the namespace... I think we need that.... and given this, maybe we should move the logic that builds this Pod skeleton into a function and just use it in both places?

Comment (akhurana001) [reply go/tasks/plugins/array/k8s/task.go]: Good catch. This is not building a pod, just name + namespace to query for the pod. There is already a common method for namespace, I changed this to use that here as well.

File: go/tasks/plugins/array/k8s/monitor.go (hunk 1 above)
Comment (akhurana001) [reply go/tasks/plugins/array/k8s/monitor.go:227]: This is cleaner as this is in-sync with the pod name which can be otherwise confusing if you are looking at k8s-logs or just investigating any issues.

**Comment by akhurana001** [comment]
@EngHabu @kumare3 PTAL

File: go/tasks/plugins/array/k8s/monitor.go (hunk 1 above)
Comment (kumare3) [reply go/tasks/plugins/array/k8s/monitor.go:227]: agreed

**Review by kumare3 — APPROVED** [review]
